great_expectations/dataset/base.py

```diff
@@ -3344,8 +3355,473 @@ def expect_column_pair_values_to_be_in_set(self,
+            minsize(integer): minimum expected file size
+            maxsize(integer): maximum expected file size
+        
+        Keyword Args: 
+            
+            result_format (str or None): \
+                Which output mode to use: `BOOLEAN_ONLY`, `BASIC`, `COMPLETE`, or `SUMMARY`.
+                For more detail, see :ref:`result_format <result_format>`.
+                
+            include_config (boolean): \
+                If True, then include the expectation config as part of the result object. \
+                For more detail, see :ref:`include_config`.
+                
+            catch_exceptions (boolean or None): \
+                If True, then catch exceptions and include them as part of the result object. \
+                For more detail, see :ref:`catch_exceptions`.
+                
+            meta (dict or None): \
+                A JSON-serializable dictionary (nesting allowed) that will be included in the output without modification. \
+                For more detail, see :ref:`meta`.
+                
+        Returns:
+            A JSON-serializable expectation result object.
+
+        Exact fields vary depending on the values passed to :ref:`result_format <result_format>` and
+        :ref:`include_config`, :ref:`catch_exceptions`, and :ref:`meta`.
+        """
+       
+        raise NotImplementedError
+   
+    def expect_file_to_exist(self,result_format=None,include_config=False,
+                             catch_exceptions=None, meta=None):
+        
+        """
+        Checks to see if a file specified by the user actually exists
+        
+        
+        Keyword Args: 
+            
+            result_format (str or None): \
+                Which output mode to use: `BOOLEAN_ONLY`, `BASIC`, `COMPLETE`, or `SUMMARY`.
+                For more detail, see :ref:`result_format <result_format>`.
+                
+            include_config (boolean): \
+                If True, then include the expectation config as part of the result object. \
+                For more detail, see :ref:`include_config`.
+                
+            catch_exceptions (boolean or None): \
+                If True, then catch exceptions and include them as part of the result object. \
+                For more detail, see :ref:`catch_exceptions`.
+                
+            meta (dict or None): \
+                A JSON-serializable dictionary (nesting allowed) that will be included in the output without modification. \
+                For more detail, see :ref:`meta`.
+                
+        Returns:
+            A JSON-serializable expectation result object.
+
+        Exact fields vary depending on the values passed to :ref:`result_format <result_format>` and
+        :ref:`include_config`, :ref:`catch_exceptions`, and :ref:`meta`.
+        """
+        
+        raise NotImplementedError
+        
+    def expect_file_unique_column_names(self, skip=0, sep=',', quoteChar='"',
```

Comment:
This seems out of place with the others--isn't this really about datatable semantics (since it's assuming a CSV?)

Reply:
Yes, I felt the same when I was transcribing it into a file expectation. It was originally written as one of the early file expectations, which is why I went ahead and wrote it as an expectations for `FileDataset`, but I think it would fit better as a `DataTable` expectation. I can take a shot at rewriting it for `PandasDataTables` and then translating it for `SqlAlchemyDataTables`.

Reply:
I rewrote this expectation as `expect_file_to_have_valid_table_header` which determines if the delimited entries on a line specified by the user are unique. In other words, for a specified line, the expectation checks to see if that line could be a valid table header with unique column names.